0-base-protocol-background.md

```diff
@@ -0,0 +1,39 @@
+# IRC-0 Background on StateChannels Protocol
+
+## Overview
+State channels is a core protocol "engine" (they call it a `wallet`) that validates and signs channel updates.
+
+It is similar to our existing CFCore module in the following ways:
+1. The core "application" lifecycle of `propose`--> `install`--> `takeAction` --> `uninstall` is retained.
+2. Data flows through an external transport layer (for inbound messages from a counterparty) or JSON rpc interface (for user-initiated inputs), gets validated by the protocol against previous state, and then gets stored in store.
+3. The output of state transitions is a set of commitments that can be put onchain in the event of a dispute.
+
+It is different in the following ways:
+1. The "application" lifecycle is named `create`--> `join`--> `update`--> `close`. The same cycle is used both for "subchannels"/virtual channels (i.e. apps), **and** for the ledger channel. This dramatically simplifies the interfaces.
+2. The core protocol handles deposit/withdraw natively. //TODO How do channel topups and partial withdraws work?
+3. There are no more multisig contracts. Instead there's a global `assetHolder` contract per asset. //TODO How do we deposit directly to an address? How do we allow counterparty to pay for withdrawal?
+4. The StateChannels wallet does not handle message dispatching and responding directly. Instead, it uses a `pushMessage` pattern whereby the implementer pushes updates into the wallet. This dramatically simplifies adding middleware for encryption, message retries, caching, etc.
+5. The protocol extends turn based updating from being only an "app"-level feature to instead be throughout the protocol. This removes the need for distributed locks, dramatically reduces the risk of states getting out of sync with each other (TODO: verify), and allows parties to receive updates entirely out of order with no problems.
```

Comment:
Channels could still get out of sync with each other, if (for example) a message is dropped from A to B. The difference is there isn't an action that B can take, because both participants believe the counterparty is the turn taker.

The real problem with channels being out of sync is that either person can initiate an update without knowing the information isn't up to date. In this paradigm, neither person would issue an update if the messages are dropped, so the out of sync issues are simplified and can mostly be handled via an intelligent messaging service.

There may still be some edgecases (like the ones we saw with the data getting truncated on restore) that we are unaware of, so at the least we should make sure a distinct error code is thrown.